Replace buttons with dropdowns, fix sizing, fix focus state

christiemolloy opened the pull request:
Fix #1183, fix #1381 , fix #1379, fix #1286

Issue: 
- The kebabs in the table were using the button component not the dropdown component.
- The dropdown toggle has the correct padding but the icon is not inheriting the line-height of its parent therefore is at 28px in height not 36px.
- Focus states for the dropdown/ option menu button were not present.

patternfly-build commented:
Deploy preview for *pf-next* ready!

Built with commit de116f50c4b9575cede3eea8f78e53ea8346f928

https://deploy-preview-1384--pf-next.netlify.com

mcoker commented:
Looks good. Curious about these nested kebabs in the compound expansion example - should these be dropdowns, too?

<img width="1010" alt="screen shot 2019-02-07 at 1 33 44 pm" src="https://user-images.githubusercontent.com/35148959/52437667-15990680-2add-11e9-8203-842099694392.png">

Also can you use unique ID's for the dropdowns? The a11y checker compares all examples for a component on a single page, so ID's between examples need to be unique.

christiemolloy commented:
@mcoker thanks for the review! I pushed changes -- added dropdowns for the nested table in compound expansion (for the time being I think its safe to assume those will be dropdowns), I added unique IDs, and I also replaced the kebab buttons in the data list with dropdowns.

mcoker commented on src/patternfly/components/AppLauncher/app-launcher.scss:
```diff
@@ -11,7 +11,15 @@
   --pf-c-app-launcher__menu--ZIndex: var(--pf-global--ZIndex--xs);
```
Looks like we need to add a normal `font-weight` to `.pf-c-app-launcher__menu-item`. There is a global style for `a` elements that uses a `semi-bold` font-weight. You can mimic the dropdown component for this change.

mcoker commented on src/patternfly/components/Dropdown/dropdown.scss:
```diff
@@ -19,6 +19,8 @@
   --pf-c-dropdown__toggle--hover--BorderBottomColor: var(--pf-global--active-color--100);
```
can you add `--pf-c-dropdown__toggle--Color: var(--pf-global--Color--100);` (then set the `color` property) so that when you toggle the dropdown in a dark theme, the toggle text updates?

And on L255, we're redefining `--pf-c-dropdown__menu-item--Color` to `var(--pf-c-dropdown__menu-item--Color)` which doesn't do anything :) Can you change that to `--pf-c-dropdown__menu-item--Color: var(--pf-c-dropdown__menu-item--hover--Color);` and then add `  --pf-c-dropdown__menu-item--hover--Color: var(--pf-global--Color--dark-100);`. That will also fix https://github.com/patternfly/patternfly-next/issues/1286

christiemolloy replied:
Great! It works now.

mcoker commented on src/patternfly/components/Dropdown/dropdown.scss:
```diff
@@ -160,6 +171,10 @@
   &.pf-m-plain {
     color: var(--pf-c-dropdown__toggle--m-plain--Color);
 
+    > i {
```
We can't reliably count on an `<i>` element here. For example, they're `<svg>` in react. I think you could change this to `> *`...

mcoker commented on src/patternfly/components/OptionsMenu/options-menu.scss:
```diff
@@ -8,6 +8,7 @@
   --pf-c-options-menu__toggle--PaddingBottom: var(--pf-global--spacer--form-element);
   --pf-c-options-menu__toggle--PaddingLeft: var(--pf-global--spacer--sm);
```
can you add a `color` to the toggle, similar to my comment about the dropdown, so that the color toggles properly in dark mode?

mcoker commented on src/patternfly/components/OptionsMenu/options-menu.scss:
```diff
@@ -132,10 +145,16 @@
       color: var(--pf-c-options-menu__toggle--m-plain--Color);
     }
 
+    > i {
```
same comment here as with dropdown.

mcoker commented on src/patternfly/components/DataList/examples/data-list-check-action-example.hbs:
```diff
@@ -18,6 +21,9 @@
       Secondary content
     {{/data-list-cell}}
-    {{#> data-list-action button--attribute='aria-labelledby="check-action-item2 check-action-action2" id="check-action-action2" aria-label="Actions"'}}{{/data-list-action}}
+    {{#> data-list-action button--attribute='aria-labelledby="check-action-item2 check-action-action2" id="check-action-action2" aria-label="Actions"'}}
```
did you mean to leave `button--attribute` on here? you might also see if you can add the `{{#> dropdown` to `data-list-action.hbs` except omit the `id` attribute and add it to `{{#> data-list-action` instead.

christiemolloy replied:
I moved `{{#> dropdown` to `data-list-action.hbs` , and the id is now on data-list-action. But my concern is that I believe the button inside the dropdown needs an ID. It takes in `dropdown-id` and concats it onto `-button` but I'm having trouble with handlebars passing the `data-list-action--ID` into the dropdown id.

If this is not possible I think I may have to add the dropdown back in otherwise it will not have a unique ID. @mcoker

mcoker replied:
You don't have `button--attribute` on any of the other `data-list-action` partials - why do you need it here? And re: the ID, I meant do `{{#> data-list-action id="dropdown-kebab-right-aligned-2"}}` - that should pass `id` in the context of `data-list-action`, which can be consumed by `{{#> dropdown}}` inside of it.

christiemolloy replied:
I had removed button--attribute not sure why it was showing up. Should be good now

mcoker commented on src/patternfly/components/Dropdown/dropdown.scss:
```diff
@@ -9,6 +9,7 @@
   --pf-c-dropdown__toggle--MinWidth: var(--pf-global--target-size--MinWidth);
```
can you also add `--pf-c-dropdown__toggle-button: var(--pf-global--color--100);` and then `color: var(--pf-c-dropdown__toggle-button);` to `pf-c-dropdown__toggle-button`? You can verify this fixes the split button dropdown in a dark theme. Currently the toggle button is black in a dark theme and it should turn white.

christiemolloy replied:
Yes working now!

mcoker commented on src/patternfly/components/Dropdown/docs/code.md:
```diff
@@ -38,7 +38,7 @@ The dropdown menu can contain either links or buttons, depending on the expected
 | `.pf-m-align-right` | `.pf-c-dropdown__menu` | Modifies to display the menu aligned to the right edge of the toggle |
```
in these docs, there is a line for

> aria-expanded=“true” | .pf-c-dropdown__menu | ...

I don’t think we need that? `aria-expanded` goes on the toggle, not the menu.

christiemolloy replied:
yeah this was also on the app launcher component so removed them both.

mcoker replied:
Can you also remove this? applied to the select dropdown which is it's own component now.

> aria-pressed="true" | button.pf-c-dropdown__menu-item | Select only The attribute aria-pressed="true" should be set programmatically to the active item. Value should be false when not selected.
-- | -- | --

mcoker commented on src/patternfly/components/OptionsMenu/docs/code.md:
```diff
@@ -24,5 +24,8 @@
 | `.pf-m-top` | `.pf-c-options-menu` | Modifies to display the menu above the toggle. |
 | `.pf-m-align-right` | `.pf-c-options-menu__menu` | Modifies to display the menu aligned to the right edge of the toggle |
 | `.pf-m-expanded` | `.pf-c-options-menu` |  Modifies for the expanded state. |
+| `.pf-m-hover` | `.pf-c-options-menu__toggle`, `.pf-c-options-menu__toggle .pf-m-plain` | Modifies for the hover state. |
```
do you need `.pf-c-options-menu__toggle .pf-m-plain` here? Seems like that's a typo, too (should be ``.pf-c-options-menu__toggle.pf-m-plain`) but it seems like listing `.pf-c-options-menu__toggle` covers it.

christiemolloy replied:
Agreed.

mcoker commented:
Great work! Left a few comments for some small things.
